### DISCUSSION TITLE
Power Button Setting Not Working

### DISCUSSION THREAD

**Discussion by R101G** [rern/rAudio#1763] [Bug Reports]
@rern 
The power button settings window does not seem able to change the config.txt entry. I set power off to 3, but config.txt showed it as 25. a manual edit fixed it.

EDIT: Turning the Power feature off and on again sets the i/o to 25. Also, the shutdown button function skips the shutdown.sh file, which normally turns off the backlight.

**Comment by R101G** [comment]
@rern 
Power button still does not shut down the local display backlight. I have -

_#! /usr/bin/bash
echo 1 > /sys/class/backlight/rpi_backlight/bl_power_

 - in shutdown.sh, which used to work. Power off from the menu works.

**Reply by rern** [reply]
Try:
```sh
# Turn off
bl_power=/sys/class/backlight/rpi_backlight/bl_power
echo 1 > $bl_power
# If not off:
chmod 666 $bl_power
echo 1 > $bl_power
```

**Reply by R101G** [reply]
@rern 
I replaced my line in shutdown.sh with this, and it did not work at all. I assume that is what you meant? The screen remained brightly lit, whereas it just glows dimly with the original line.

**Reply by R101G** [reply]
@rern 
I misunderstood your message, but when I corrected the entry it behaved as before. The system shuts down, but leaves a dim glow on the display. The menu shutdown blanks it completely. If I use the bl_power command from the command prompt, it also blanks completely, so something about  boot/shutdown.sh must be different.

**Reply by Bautar** [reply]
Hello - Related issue (maybe?) here. This happened in the past and you corrected it, but I’m rebuilding a machine (pi 0-W), and when I navigate to <System> /GPIO Devices/Power Button, the menu to set the “On” pin is blank/non-funcional. Off and Power are available but without “On”, the feature doesn’t function:
![image](https://github.com/user-attachments/assets/9ebb99a7-794c-473c-8d80-932eb2069f9e)

**Comment by R101G** [comment]
@Bautar 
I have just checked, and my default is showing 5 for on, which is the pin number, rather than the i/o, which is confusing. I think you are right that this is similar to something we have seen before. It was possible to set both on and off to 3, so that it toggles.

**Reply by Bautar** [reply]
Thanks, Rern. Unfortunately I can’t change “on” to 3 (or anything) because the drop-down box is unavailable. I did try changing “off” to 3, but the on box was still missing. 
I looked back & the previous report was ID’d as #1650 and you did come up with a fix which corrected the problem, from April 2 (my apologies; I never sent a “Thank you”…Thank you!)
I do have another Pi0-W build which has/had? a functioning power button, but it’s on loan to a friend. As soon as I get my hands on it I’ll see if the button is working & maybe even load the configuration to my new machine to see if it works.
Thanks- Ken

**Comment by R101G** [comment]
@rern 
It looks as if copying my shutdown.sh file to /usr/lib/systemd/system-shutdown/ solves the backlight not turning off problem. The pin setting might be a different matter.

**Reply by Bautar** [reply]
I was able to restore a backup onto my new machine which had the power button functioning fine now. I have a single button wired only to pin 5 & it turns the machine both on and off. This screenshot shows there is still a problem: “On” does not have a drop-down arrow so it can’t be changed, and it shows it is set to “3” for “On” & 5 for “Off”. 
Mom all set & there’s probably more important things to work on!
Thanks for your help!
<img width="414" alt="image" src="https://github.com/user-attachments/assets/ccb924df-7db6-42e8-87d2-79622eab08b6">

**Reply by R101G** [reply]
The setting used to refer to the GPIO (3) rather than the pin number (5). The feature obviously got messed up by one of the upgrades, but not necessarily a recent one. I had to edit the /boot/config.txt file entry to get the shutdown working, as the "off" setting was not being saved correctly.
The really stupid thing is that pins 3 and 5 are also used for I2c, so it did not occur to the Pi designers that anyone would want both I2c and a power toggle.

**Comment by R101G** [comment]
@rern 
I have downloaded and installed a clean new image, and still had to add `dtoverlay=gpio-shutdown,gpio_pin=3` to config.txt manually. The menu setup just does not work. (Likewise, the wifi setup from the menu did not accept a fixed IP without editing the netctl file manually.)
